solution/zenko-version-cr.yaml

```
@@ -0,0 +1,48 @@
+---
+apiVersion: zenko.io/v1alpha1
+kind: ZenkoVersion
+metadata:
+  name: 1.1-dev
+spec:
+  imagePullPolicy: Always
+  versions:
+    cloudserver:
+      image: registry.scality.com/zenko-dev/cloudserver
+      tag: latest-8.1.2
+    backbeat:
+      image: registry.scality.com/zenko-dev/backbeat
+      tag: 161b9c1
+    blobserver:
+      image: registry.scality.com/sf-eng/blobserver
+      tag: 1.0.8
+    jabba:
+      image: registry.scality.com/sf-eng/jabba
+      tag: 940d
+    secureChannelProxy:
+      image: registry.scality.com/zenko-dev/cloudserver
+      tag: 4332d0a
+    localData:
+      image: zenko/cloudserver
+      tag: 8.1.15
+    metrics:
+      image: registry.scality.com/zenko-dev/cloudserver
+      tag: 4332d0a
+    s3utils:
+      image: zenko/s3utils
+      tag: 1.2
+    zookeeper:
+      image: pravega/zookeeper
+      tag: 0.2.4
+    kafka:
+      image: wurstmeister/kafka
+      tag: 2.12-2.3.0
+    shell:
+      image: busybox
+      tag: 1.31.0-glibc
+    mongodb:
+      image: mongodb
+      tag: 4.0
+    redis:
+      image: redis
+      tag: 5.0.3-v1
```

the `mongodb` and `redis` should have the following structure
```suggestion
    mongodb: 4.0
    redis: 5.0.3-v1
```

---

Hmm, are these not equivalent? My bash parsing uses the image: and tag: lines to pull from, these two were special cases I was trying to unspecial. Not a huge deal though.

---

Actually, regardless I dont seem to be able to pull mongo or redis of these tags.
* `docker pull mongodb:4.0`
* `docker pull mongodb:4.0-v1` ( this was the actual image used )
* `docker pull redis:5.0.3-v1`

Am I missing something here?

---

they aren't. the `mongodb` and `redis` services will be managed by the `kubedb` operator, and those have a different way of defining versions.

---

the mongodbversion has the following images:
```
apiVersion: catalog.kubedb.com/v1alpha1
kind: MongoDBVersion
metadata:
...
  name: 4.0-v1
spec:
  db:
    image: kubedb/mongo:4.0-v1
  exporter:
    image: kubedb/mongodb_exporter:v1.0.0
  initContainer:
    image: kubedb/busybox
  podSecurityPolicies:
    databasePolicyName: mongodb-db
    snapshotterPolicyName: mongodb-snapshot
  tools:
    image: kubedb/mongo-tools:4.0-v1
  version: 4.0.5

```

---

the redisversion:
```
apiVersion: catalog.kubedb.com/v1alpha1
kind: RedisVersion
metadata:
...
  name: 5.0.3-v1
spec:
  db:
    image: kubedb/redis:5.0.3-v1
  exporter:
    image: kubedb/redis_exporter:v0.21.1
  podSecurityPolicies:
    databasePolicyName: redis-db
  version: 5.0.3

```

---

oh ok, I can accept your change but won't I still need to pre-pull and save the mongo and redis docker images in the ISO anyway?

---

yeah, you will still need to do so.

---

oh nice, let me get this in there.